src/rust/ide/view/graph-editor/src/component/node/error.rs

```diff
@@ -1,35 +1,41 @@
 //! Contains a struct definition for error information on nodes.
 use crate::prelude::*;
 
-use crate::component::node::visualization;
+use crate::builtin::visualization::native::error as error_visualization;
+
+use serde::Deserialize;
+use serde::Serialize;
 
 
 
 // =============
 // === Error ===
 // =============
 
-/// Error information to be displayed on a node.
-///
-/// Note:
-/// Tis is a dummy implementation that can and should be extended once we have the proper
-/// error information from the language server.
-/// See [issue #1026](https://github.com/enso-org/ide/issues/1026) for more information.
-#[derive(Clone,Debug)]
-pub struct Error {
-    /// The error message to show on the node.
-    pub message:String,
-}
+/// An error kind.
+#[allow(missing_docs)]
+#[derive(Clone,Copy,Debug,Deserialize,Serialize)]
+pub enum Kind {Panic,Dataflow}
 
-impl From<Error> for visualization::Data {
-    fn from(error:Error) -> visualization::Data {
-        let content = serde_json::Value::String(error.message).into();
-        visualization::Data::Json {content}
-    }
+/// Additional error information (beside the error value itself) for some erroneous node.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Error {
+    pub kind : Immutable<Kind>,
+    /// An error message overriding the error visualization data. Should be set in cases when the
+    /// visualization won't work (e.g. in case of panics).
+    pub message : Rc<Option<String>>,
+    /// Flag indicating that the error is propagated from another node visible on the scene.
+    pub propagated : Immutable<bool>,
```

Comment:
I think you dont need `Imutable` here

Reply:
I wanted it to be clone_ref-able, because this structure is heavily propagated trough the FRP network. And deriving CloneRef requires wrapping in immutable.